include/arch/arm/arch.h

```diff
@@ -216,6 +216,11 @@ extern "C" {
 #define K_MEM_PARTITION_P_RWX_U_RWX	(P_RW_U_RW)
 #define K_MEM_PARTITION_P_RWX_U_RX	(P_RW_U_RO)
 #define K_MEM_PARTITION_P_RX_U_RX	(P_RO_U_RO)
+
+#define K_MEM_PARTITION_IS_WRITABLE(attr) \
+	((attr) & (P_RW_U_RW | P_RW_U_RO | P_RW_U_NA))
+#define K_MEM_PARTITION_IS_EXECUTABLE(attr) (1)
```

Comment:
@AdithyaBaglody  this is the non-PAE case so all pages are executable

Reply:
this is the case for the ARM MPU. yes this doesn't quite look right, shouldn't be a hardcoded 1

Reply:
There are no P_aa_U_bb where aa or bb contains an X in any of the uses above it, so I just assumed that the ARM MPU does not have a `X` or `NX` bit.  That's why there's a `CONFIG_ARCH_HAS_EXECUTABLE_BIT` option that the NXP MCU defines to `y`.  Because of this, `ensure_w_xor_x()` is a no-op on systems with an ARM MPU.

Reply:
Indeed, you're right.  I've fixed my patch and sent it again.